src/main/java/org/terasology/equipment/system/EquipmentEffectsSystem.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.equipment.system;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.alterationEffects.AlterationEffect;
+import org.terasology.alterationEffects.regenerate.RegenerationAlterationEffect;
+import org.terasology.alterationEffects.regenerate.RegenerationComponent;
+import org.terasology.alterationEffects.speed.*;
+import org.terasology.context.Context;
+import org.terasology.entitySystem.Component;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.equipment.component.effects.RegenEffectComponent;
+import org.terasology.equipment.component.effects.StunEffectComponent;
+import org.terasology.equipment.component.EquipmentComponent;
+import org.terasology.equipment.component.EquipmentEffectComponent;
+import org.terasology.equipment.event.EquipItemEvent;
+import org.terasology.equipment.event.UnequipItemEvent;
+import org.terasology.logic.health.BeforeDamagedEvent;
+import org.terasology.registry.In;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+@RegisterSystem
+public class EquipmentEffectsSystem extends BaseComponentSystem {
+
+    @In
+    private Context context;
+
+    /**
+     * Maps the EquipmentEffectComponents to their corresponding EffectComponents so that
+     * 1. the system knows which EquipmentEffectComponents to look out for
+     * 2. the system knows which EffectComponents to remove from the entity on unequip
+     */
+    private Map<Class, Class<? extends  Component>> effectComponents = new HashMap<>();
+
+    @Override
+    public void initialise() {
+        effectComponents.put(StunEffectComponent.class, StunComponent.class);
+        effectComponents.put(RegenEffectComponent.class, RegenerationComponent.class);
```

Comment:
Consider extracting this into a separate method so that other modules can add their own effects.